server/src/main/java/io/crate/protocols/postgres/types/NumericType.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.protocols.postgres.types;
+
+import io.netty.buffer.ByteBuf;
+
+import javax.annotation.Nonnull;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.MathContext;
+import java.nio.charset.StandardCharsets;
+
+class NumericType extends PGType<BigDecimal> {
+
+    static final int OID = 1700;
+
+    private static final int TYPE_LEN = -1;
+    private static final int TYPE_MOD = -1;
+
+    public static final NumericType INSTANCE = new NumericType();
+
+    private NumericType() {
+        super(OID, TYPE_LEN, TYPE_MOD, "numeric");
+    }
+
+    @Override
+    public int typArray() {
+        return PGArray.NUMERIC_ARRAY.oid();
+    }
+
+    @Override
+    public String typeCategory() {
+        return TypeCategory.NUMERIC.code();
+    }
+
+    @Override
+    public String type() {
+        return Type.BASE.code();
+    }
+
+    @Override
+    public int writeAsBinary(ByteBuf buffer, @Nonnull BigDecimal value) {
+        buffer.writeInt(TYPE_LEN);
```

Comment:
@seut I couldn't finish this part on time, so such as I am not here, I am handling it over to you :( 
I've been trying to come up with some solution that would reflect the numeric_send method   https://github.com/postgres/postgres/blob/7538708394e7a70105a4e601e253adf80f47cca8/src/backend/utils/adt/numeric.c#L1030-L1049 but couldn't make it work (testing with asyncgpg). I've been writing num digits, weight, sign, scale and each digit as short (int16), but it seems that it is more complicated than that, see  https://github.com/cockroachdb/cockroach/blob/c2b4fff9ce4a0b80f74dd39a5e45eccd15c1b933/pkg/sql/pgwire/pgwirebase/encoding.go#L600-L637 and https://github.com/cockroachdb/cockroach/blob/c2b4fff9ce4a0b80f74dd39a5e45eccd15c1b933/pkg/sql/pgwire/pgwirebase/encoding.go#L550-L662